providers/summary/scancode.js

```diff
@@ -28,22 +28,30 @@ class ScanCodeSummarizer {
     const result = {}
     this.addDescribedInfo(result, coordinates, harvested)
     const buckets = this.computeFileBuckets(harvested.content.files, facets)
-    set(result, 'licensed.facets', {})
-    const facetsObject = get(result, 'licensed.facets')
-    const declaredLicenses = new Set()
+    let declaredLicenses = new Set()
+    let packageLicenses = new Set()
+    for (const fileIdx in harvested.content.files) {
+      const file = harvested.content.files[fileIdx]
+      if (!file.path.toLowerCase().includes('/')) {
+        const asserted = get(file, 'packages[0].asserted_licenses')
+        if (asserted && packageLicenses.size === 0) this._addArrayToSet(asserted, packageLicenses, license => license.license || license.spdx_license_key)
+        declaredLicenses = this._summarizeLicenseFile(file)
+        if (declaredLicenses.size > 0) break
+      }
+    }
+    set(result, 'licensed.declared', this._toExpression(declaredLicenses || packageLicenses))
+    result.files = []
     for (const key in buckets) {
-      facetsObject[key] = this.summarizeLicenseInfo(buckets[key])
-      this.summarizeDeclaredLisense(buckets[key], declaredLicenses)
+      set(result, `licensed.facets.${key}`, this.summarizeLicenseInfo(buckets[key]))
+      this.summarizeFileInfo(buckets[key], result.files, key)
```

Comment:
Should not have to pass `result.files` here. `files` should be set by the other loop that is going over all the files. That is independent of the facetization (if any).

Reply:
I think there was a misunderstanding in my last commit ("Look at last component of path for license/package, made code more modular", e8035f3c48722aa75389b612df2e857345d685b2). It seems that each file needs to be tagged that facets it belongs to. It seems that I have to compute per file info in this loop since I will know exactlu which facet I am looping over. If I summarize per file info in its own loop, then how will each file know what facet it belongs to. I think I am not understanding how you know which facet a file belongs to.